### DISCUSSION TITLE
what is a wrapped.com.ibm.icu.icu4j_3.8.0.jar dependency for?

### DISCUSSION THREAD

**Discussion by MuriloIPM** [eclipse-birt/birt#1107] [Q&A]
On my Windows machine running my application after updating to birt 4.10, my dates with the pt-br location with the SHORT type "11/09/2022" (Correct), already when I started the war in the docker image "tomcat:9.0 -jdk11" I noticed that my dates came out of the following format "09/11/22", after breaking my head and making several changes to the location of the image, I deleted this jar and returned to the correct format.

What is wrapped.com.ibm.icu.icu4j_3.8.0.jar for? because birt already uses another jar com.ibm.icu_70.1.0.jar

**Comment by speckyspooky** [comment]
I started a short research on your mentioned jar-files and BIRT use with ICU4J 70.1.0 a modern version of ICU4J - Release date: 2021-10-27, https://icu.unicode.org/home.

The other version seems to be a very eraly version which is much older like the BIRT versoin. I figured out a version summary there - https://frontbackend.com/maven/artifact/com.ibm.icu/icu4j/3.8.

It could be that the change which you figured out before cause the changed behavior on your side.

**Reply by wimjongman** [reply]
@speckyspooky do we still ship the old icu version? I think most of the functions from ICU are available in the JVM these days but it is probably quite some work to replace it.

**Reply by MuriloIPM** [reply]
But I still don't understand why keep the jar "wrap.com.ibm.icu.icu4j_3.8.0.jar" along with the latest one which would be " jar com.ibm.icu_70.1.0.jar". Wouldn't you just delete the old jar and keep the latest one?

**Reply by speckyspooky** [reply]
@wimjongman We ship one of the newest versions, so from ym point of few we are save with it.
But currentley I'm a little bit confused, because I verified my BIRT 4.10 installations (original download from our download page) and I see 3 versions of ICU.
My old installations 4.6, 4.8, 4.9 have only 1 version of it. Is there an reason that we have 3 versions added.
Screens attached:

BIRT 4.10: eclipse all-in-one designer 
![2022-11-11 20_28_26-plugins](https://user-images.githubusercontent.com/41593722/201416046-afad22a8-dc80-4247-bde8-4cb629af4b73.png)

BIRT 4.10: OSGI-version
![2022-11-11 20_35_16-lib](https://user-images.githubusercontent.com/41593722/201416933-5e581052-d238-4fec-aa60-7e67491f1d73.png)

**Reply by TheLaserNinja** [reply]
Weird. Not surprising considering the number of versions we've gone through over the past. We've had that in there since the beginning. So we might have the wrap one present for any interfaces or class definitions that radically changed. This was always the crux of the problem with this library IIRC. 

I looked through the release history, and 67 was released at least as of 03-2020. So, not sure why we have duplicate 67 releases in the library path. Probably just a build scirpt issue where one of the plugins is date-stamping the JAR file. But the better question is there really that much of a difference between 70 and 67 that would prevent us from not just going with a newer version?

**Reply by MuriloIPM** [reply]
Birt 4.10 Runtime there are two jars referring to icu
![image](https://user-images.githubusercontent.com/113636465/202174686-1cd3850e-bf42-4fdb-9caf-da83ed3f4393.png)
![image](https://user-images.githubusercontent.com/113636465/202174792-120afde9-cfc6-4c96-889c-e5f8e353ebaa.png)

**Comment by TheLaserNinja** [comment]
The IBM ICU package is IBM's library for providing unicode, localization, and date formatting to Java based on the constantly evolving standards. It was part of BIRT since roughly the beginning. It's used mostly for the date formatting functions, so that would line up with what you are seeing. It's caused us a few headaches in the past. But it's an incredibly powerful library for what it does.

**Reply by wimjongman** [reply]
I did a search on all our XML's and we mentioned some different versions [1]. Someone should run through all the dependencies and see if this causes the duplication. We do not need the 67 version, I would say.

[1] https://github.com/eclipse/birt/search?l=XML&p=5&q=com.ibm.icu

**Reply by speckyspooky** [reply]
Hello together,
short summary is seems to be different built scripts exists for BIRT, because we habe 2 situations currently.

The BIRT rcp-package, designer-all-in-one and birt-orsgi includes 3 version of icu (70, 67, 67..v).
The BIRT runtime-packahe includes 2 versions (70 and very old 3.8.0) -> @MuriloIPM, I confirm it.

It would be great if we could use the same version 70 my be version 72 from October this year like an update für 4.11
to get the same situation on all packages.

@wimjongman would it be possible to synchronize all in from your side?
It would be great and very helpful.

**Reply by wimjongman** [reply]
Yes that sounds great. Can you make a PR?
